core/libgodot/libgodot.h

```diff
@@ -0,0 +1,73 @@
+/**************************************************************************/
+/*  libgodot.h                                                            */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef LIBGODOT_H
+#define LIBGODOT_H
+
+#if defined(LIBRARY_ENABLED)
+
+#if defined(WINDOWS_ENABLED) | defined(UWP_ENABLED)
+#define LIBGODOT_API __declspec(dllexport)
+#elif defined(ANDROID_ENABLED)
+#include <jni.h>
+#define LIBGODOT_API JNIEXPORT
+#else
+#define LIBGODOT_API
+#endif
+
+#include "core/extension/gdextension_interface.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+void libgodot_init_resource();
+
+void libgodot_project_settings_load(void *project_settings);
+
+void libgodot_scene_load(void *scene);
+
+bool libgodot_is_scene_loadable();
+
```

Comment:
bool does not exist in plain C interfaces.

Reply:
Indeed, C99 requires stdbool.h to be included in order for the `bool` type to be defined.
I suggest adding this to the top of the file to permit pure C -std=c99 code to use the header:
```cpp
#ifndef __cplusplus
#include <stdbool.h>
#endif
```
I don't think there is a need to avoid use of bool because c99 support is probably widespread by now.

Reply:
I was thinking in terms of the whole godot api where bool may be used in many places...
But if it's just that one function, it might be easier to save the hassle and return int and avoid needing stdbool